### PR TITLE
`merkledb` -- encode lengths as uvarints

### PR DESCRIPTION
**Pull request by danlaine**
## Why this should be merged

These lengths can never be negative.

## How this works

`Varint` --> `Uvarint`

## How this was tested

Existing UT

### CODE REVIEW

**Review by richardpringle — APPROVED** [review]

File: x/merkledb/codec.go [hunk 1]
```diff
@@ -221,18 +220,15 @@ func (*codecImpl) decodeBool(src *bytes.Reader) (bool, error) {
 	}
 }
 
-func (c *codecImpl) encodeInt(dst *bytes.Buffer, value int) {
```
Comment (danlaine) [inline x/merkledb/codec.go:224]: We only encode non-negative ints so this was replaced with `encodeUint64`

File: x/merkledb/codec.go [hunk 2]
```diff
@@ -221,18 +220,15 @@ func (*codecImpl) decodeBool(src *bytes.Reader) (bool, error) {
...
-	c.encodeInt64(dst, int64(value))
-}
-
-func (*codecImpl) decodeInt(src *bytes.Reader) (int, error) {
+// decodeUint decodes a uvarint from [src] and returns it as an int.
+func (*codecImpl) decodeUint(src *bytes.Reader) (int, error) {
```
Comment (danlaine) [inline x/merkledb/codec.go:224]: Arguably this should return a `uint` but a bunch of places where we use this, we want it as an `int`, so idk

Comment (dboehm-avalabs) [reply x/merkledb/codec.go:224]: do we want to just update all of those variables to be uints instead?

**Review by dboehm-avalabs — APPROVED** [review]

File: x/merkledb/codec.go [hunk 3]
```diff
@@ -221,25 +218,19 @@ func (*codecImpl) decodeBool(src *bytes.Reader) (bool, error) {
 	}
 }
 
-func (c *codecImpl) encodeInt(dst *bytes.Buffer, value int) {
-	c.encodeInt64(dst, int64(value))
-}
-
-func (*codecImpl) decodeInt(src *bytes.Reader) (int, error) {
```
Comment (danlaine) [inline x/merkledb/codec.go:228]: Changed to `decodeUint` since we always read uint64s

**Review by xinifinity — APPROVED** [review]

**Pull request merged**